src/main/main.js

```diff
@@ -82,51 +88,41 @@ let isSecondInstance = electron.app.makeSingleInstance((commandLine, workingDire
 function onReady(app) {
 	return async () => {
 		global.__static = __static;
-
 		if (isSecondInstance) {
+			log.warn('another instance of the app is running, quiting');
 			electron.app.quit();
 			return;
 		}
-
-		if (process.env.NODE_ENV !== 'development' && process.env.MODE !== 'test') {
+		if (!isDevMode() && !isTestMode()) {
 			// Initate auto-updates
 			const { appUpdater } = require('./autoupdater');
 			appUpdater();
 		}
-
-		const initDb = require('./services/knex').init;
-
-		await initDb();
-		await crashReportService.startCrashReport();
+		await db.init();
+		if (!isDevMode() && !isTestMode()) {
+			await crashReportService.startCrashReport();
+		}
 		const store = configureStore(global.state, 'main');
 		try {
 			store.dispatch(localeUpdate('en'));
 			store.dispatch(fiatCurrencyUpdate('USD'));
 		} catch (e) {
-			log.error(e);
+			log.error('locale initi error %s', e);
 		}
 		app.config.userDataPath = electron.app.getPath('userData');
 
-		const PriceService = require('./controllers/price-service');
-
-		const AirtableService = require('./controllers/airtable-service');
-
-		const LedgerService = require('./controllers/ledger-service')();
-		electron.app.ledgerService = new LedgerService();
+		const priceService = (electron.app.priceService = new PriceService());
 
+		electron.app.web3Service = new Web3Service();
```

Comment:
We should get away from attaching all of these dependencies to the `electron` object.

Reply:
@dfcowell we have a plan already with @rodrigopavezi how to manage global dependancies, we'll do it in a separate issue